executor.py

```diff
@@ -1,7 +1,152 @@
+__copyright__ = 'Copyright (c) 2020-2021 Jina AI Limited. All rights reserved.'
+__license__ = 'Apache-2.0'
+
+from copy import deepcopy
+from typing import Iterable, Dict, Optional
+
+import imagehash
+import numpy as np
+from PIL import Image
 from jina import Executor, DocumentArray, requests
+from jina.logging.logger import JinaLogger
+
+HASH_TYPE = ['perceptual', 'average', 'difference', 'wavelet']
```

Comment:
It's better to direct use the function name so that we avoid the translation effort.